Helper/Order.php

```diff
@@ -498,21 +498,23 @@ private function deleteRedundantQuotes($quote) {
     public function saveUpdateOrder($reference, $frontend = true, $bolt_trace_id = null)
     {
         $transaction = $this->fetchTransactionInfo($reference);
-        $incrementId = $transaction->order->cart->display_id;
-
-        // Load quote from entity id
-        $quoteId = $transaction->order->cart->order_reference;
-        $quote = $this->cartHelper->getQuoteById($quoteId);
 
-        if (!$quote || !$quote->getId()) {
-            throw new LocalizedException(__('Unknown quote id: %1', $quoteId));
-        }
+        // get order id and immutable quote id stored with transaction
+        list($incrementId, $quoteId) = explode(' / ', $transaction->order->cart->display_id);
 
         // check if the order exists
         $order = $this->cartHelper->getOrderByIncrementId($incrementId);
 
+        // Load (immutable) quote from entity id
+        $quote = $this->cartHelper->getQuoteById($quoteId);
+
         // if not create the order
```

Comment:
nit: update this comment or re-arrange statements - I think this is continued sentence from "// check if the order exists" above

Reply:
re-arranged